include/tesla.hpp

```diff
@@ -1351,47 +1412,72 @@ namespace tsl {
              * 
              * @return X position
              */
-            virtual inline s32 getX() final { return this->m_x; }
+            inline s32 getX() { return this->m_x; }
             /**
              * @brief Gets the element's Y position
              * 
              * @return Y position
              */
-            virtual inline s32 getY() final { return this->m_y; }
+            inline s32 getY() { return this->m_y; }
             /**
              * @brief Gets the element's Width
              * 
              * @return Width
              */
-            virtual inline s32 getWidth()  final { return this->m_width;  }
+            inline s32 getWidth() { return this->m_width;  }
             /**
              * @brief Gets the element's Height
              * 
              * @return Height
              */
-            virtual inline s32 getHeight() final { return this->m_height; }
+            inline s32 getHeight() { return this->m_height; }
+
+            inline s32 getTopBound() { return this->getY(); }
+            inline s32 getLeftBound() { return this->getX(); }
+            inline s32 getRightBound() { return this->getX() + this->getWidth(); }
+            inline s32 getBottomBound() { return this->getY() + this->getHeight(); }
+
+            /**
+             * @brief Check if the coordinates are in the elements bounds
+             * 
+             * @return true if coordinates are in bounds, false otherwise
+             */
+            bool inBounds(s32 touchX, s32 touchY) {
+                return touchX >= this->getLeftBound() && touchX <= this->getRightBound() && touchY >= this->getTopBound() && touchY <= this->getBottomBound();
+            }
+            /**
+             * @brief Check if the coordinates are outside the elements bounds
+             * 
+             * @return true if coordinates are out of bounds, true otherwise
+             */
+            bool outOfBounds(s32 touchX, s32 touchY) {
```

Comment:
Isn't this just `!inBounds(...)`? Having two functions for checking opposites doesn't seem to make sense a lot, especially since outOfBounds just does the inverse check of inBounds instead of just calling that function and negating it's result. Also the name is kinda misleading imo.